File: yongkeun/6week/같은숫자는싫어.js
function solution(arr) {
  var answer = [];
  let a = arr[0];
  answer.push(a); //0번 index는 그냥 넣어줌
  for (let i = 1; i < arr.length; i++) {
    if (a === arr[i]) {
      //값이 같으면
      a = arr[i]; //a에 값을 할당, 이전 값이 a에 들어가서 이전 값과 현재 값을 비교하게 됨
    } else {
      //값이 같지 않으면
      answer.push(arr[i]); //push
      a = arr[i]; //이전 값을 넣어줌
    }
  }
  return answer;
}

File: dongmin/week6/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public List<Integer> solution(int []arr) {      //두 개 뽑아서 더하기 소스와 마찬가지로 반환형을 바꿨다.
        List<Integer> answer = new ArrayList<>();   //값이 몇개가 나올지 모르므로 동적으로 배열 크기를 조절하기 위해 ArrayList 사용
        answer.add(arr[0]);     //첫번째 인덱스값을 넣어준다.
        for (int i = 1; i < arr.length; i++) {  //반복문이 1부터 시작하기 때문에 위에서 첫번째 인덱스 값을 넣어줌
            if(arr[i-1] != arr[i])    //i를 0부터 시작하면 인덱스 값이 오버 된다.
                answer.add(arr[i]);
        }
        return answer;
    }
}